### PR TITLE
[RU]UniComics

### PR DESCRIPTION
**Pull request by e-shl**
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val document = client.newCall(GET(baseUrl + PATH_URL + manga.url, headers)).execute().asJsoup()
+        val pages = mutableListOf(1)
+        val dataStrArray = document
+            .toString()
+            .substringAfter("new Paginator(")
+            .substringBefore(");</script>").split(", ")
+        (1..dataStrArray[1].toInt()).map { i ->
+            pages.add(i)
+        }
```
Comment (alessandrojean) [inline src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt:119]:
```suggestion
        pages += (1..dataStrArray[1].toInt()).toList()
```

File: src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    override fun chapterFromElement(element: Element): SChapter {
+        val urlElement = element.select(".button.online a").first()
+        val chapter = SChapter.create()
+        element.select(".list_title").first().text().let {
+            chapter.name = it
+            chapter.chapter_number = 0F
+            if (it.contains(" №")) {
+                chapter.name = it.substringAfterLast(" ")
+                chapter.chapter_number = it.substringAfter(" №").toFloat()
+            }
```
Comment (alessandrojean) [inline src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt:146]:
```suggestion
            if (it.contains(" №")) {
                chapter.name = it.substringAfterLast(" ")
                chapter.chapter_number = it.substringAfter(" №").toFloatOrNull() ?: -1f
            }
```

File: src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = mutableListOf<Page>()
+        val dataStrArray = document
+            .toString()
+            .substringAfter("new Paginator(")
+            .substringBefore(");</script>").split(", ")
+        (1..dataStrArray[1].toInt()).map { i ->
+            pages.add(Page(i, document.location() + "/$i"))
+        }
+        return pages
```
Comment (alessandrojean) [inline src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt:161]:
```suggestion
        val dataStrArray = document.toString()
            .substringAfter("new Paginator(")
            .substringBefore(");</script>")
            .split(", ")

        return (1..dataStrArray[1].toInt()).mapIndexed { i, page ->
            Page(i, document.location() + "/$page")
        }
```

**Review by alessandrojean — CHANGES_REQUESTED** [review]

File: src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    override fun popularMangaFromElement(element: Element): SManga {
+        val manga = SManga.create()
+        manga.thumbnail_url = element.select(".left_comics img").first().attr("src").replace(".jpg", "_big.jpg")
+        element.select("a").first().let {
+            manga.setUrlWithoutDomain(it.attr("href").substringAfter(PATH_URL))
+        }
+        manga.title = element.select(".list_title_en").first().text()
+        return manga
+    }
+    override fun popularMangaParse(response: Response): MangasPage {
```
[the remark this answers is not in the document]
Comment (e-shl) [reply src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt:78]: Without it, page switching/nextpage only works in j2k

[the remark this answers is not in the document]
Comment (alessandrojean) [reply src/ru/unicomics/src/eu/kanade/tachiyomi/extension/ru/unicomics/UniComics.kt:78]: Ah, I understand what you're doing now. Can keep the function, no problem.

**Review by alessandrojean — APPROVED** [review]

**Pull request merged**